pkg/dns/dns.go

```diff
@@ -500,13 +502,20 @@ func (kd *KubeDNS) fqdn(service *v1.Service, subpaths ...string) string {
 
 func (kd *KubeDNS) newPortalService(service *v1.Service) {
 	subCache := treecache.NewTreeCache()
-	recordValue, recordLabel := util.GetSkyMsg(service.Spec.ClusterIP, 0)
-	subCache.SetEntry(recordLabel, recordValue, kd.fqdn(service, recordLabel))
+	clusterIPs := util.GetClusterIPs(service)
+
+	for _, ip := range clusterIPs {
```

Comment:
Let's factor this into its own method

Reply:
inside of the for loop

Reply:
To be honest, I think that the loop size and nestedness is manageable here without much need of moving the code out.
If anything I can do that in the next PR.